### DISCUSSION TITLE
Azure Entra ID auth

### DISCUSSION THREAD

**Discussion by petrkr** [mydumper/mydumper#1411] [Q&A] [answered]
As Azure support "one time" token as mysql password, there is need to replace password after expire.

I want know, before I will check code itself, if myloader/mydumper open connections at beginning or dynamically during long period load/dump.

If token expires, existing connections are NOT dropped, but if loader/dumper will try open new connection (or reconnect), it will fail with incorrect password already.

If it does not support, would be nice to support reading password for example from file (where will be actual valid token)

Ref: https://learn.microsoft.com/en-us/azure/mysql/single-server/how-to-configure-sign-in-azure-ad-authentication

**Comment by davidducos** [comment]
Hi @petrkr, 
By default mydumper and myloader opens connection at the begining.
We implemented on mydumper reconnection on the last prerelease when there is a failure, but only when there backup might not be consistent.
On myloader, the reconnection has been implemented a few release back when there is a failure.

**Reply by petrkr** [reply]
Thanks for reply, so in case no-error it will open all parallel connections at beginning, but there is no guaranteed it will NOT reconnect after token expiration, which will lead to fail.

How 'hard' will be implementation of read password from file instead arguement? Like if -p argu 'is file' /os.isfile()' or something, then use content and on reconnect re-read file content.

Thanks.
We trying rid of static passwords and migrate to those one-time tokens.

If I can help somehow, let me know (testing or like that)

**Reply by davidducos** [reply, marked as answer]
Hi @petrkr, if you use --defaults-extra-file with [client] and you modify `password=` with the token, then the new connection will use that password.
So, what you can do, let say that your token expire every 5 minutes, then create a script that update the configuration file with the password.
